### PR TITLE
[coin] x64-windows-statid-md fixes

### PR DESCRIPTION
**Pull request by ankurvdev**
Fix the coin port to build with x64-windows-static-md

- What does your PR fix? Fixes #

- Which triplets are supported/not supported? Have you updated the CI baseline?

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

**Comment by ankurverma85** [comment]
> Please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

Done. Please review and merge

File: ports/coin/portfile.cmake [hunk 1]
```diff
@@ -18,6 +18,12 @@ else()
     set(COIN_BUILD_SHARED_LIBS ON)
 endif()
 
+if(VCPKG_CRT_LINKAGE STREQUAL dynamic)
+    set(COIN_BUILD_MSVC_STATIC_RUNTIME OFF)
+elseif(VCPKG_CRT_LINKAGE STREQUAL static)
+    set(COIN_BUILD_MSVC_STATIC_RUNTIME ON)
+endif()
```
Comment (JackBoosY) [inline ports/coin/portfile.cmake:25]: You can see above that `COIN_BUILD_MSVC_STATIC_RUNTIME` has been changed, maybe you need to delete them from the judgment of `VCPKG_LIBRARY_LINKAGE`?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/coin/portfile.cmake (hunk 1 above)
Comment (ankurverma85) [reply ports/coin/portfile.cmake:23]: Yes. Thanks for highlighting

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**